modules/core/src/main/java/org/locationtech/jts/geom/ExtendableCoordinateSequence.java

```diff
@@ -0,0 +1,425 @@
+
+/*
+ * Copyright (c) 2018 Felix Obermaier.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+package org.locationtech.jts.geom;
+
+import org.locationtech.jts.geom.impl.CoordinateArraySequence;
+import org.locationtech.jts.geom.impl.PackedCoordinateSequence;
+
+import java.io.Serializable;
+
+/**
+ * A wrapper around a coordinate sequence that always
+ * ensures that a sequence is large enough to store ordinate
+ * values.
+ */
+public class ExtendableCoordinateSequence implements CoordinateSequence, Serializable {
```

Comment:
This class name kind of conflicts with [ExtendedCoordinateSequence.java](https://github.com/locationtech/jts/blob/master/modules/example/src/main/java/org/locationtech/jtsexample/geom/ExtendedCoordinateSequence.java) which we have noticed is in common use by downstream projects

Reply:
Perhaps we could use the word delegate since it is delegating storage to another csFactory?

Reply:
- `DynamicResizingCoordinateSequence`?
- `AutoResizingCoordinateSequence`?
- `ResizingCoordinateSequence`?
- `ExpandableCoordinateSequence`?
- `ExpandingCoordinateSequence`?
- `VariableSizeCoordinateSequence`?